skyve-war/src/main/java/modules/admin/User/UserService.java

```diff
@@ -267,4 +274,126 @@ public void evictUserProxy(UserExtension bean) {
 			p.evictSharedCachedBean(UserProxy.MODULE_NAME, UserProxy.DOCUMENT_NAME, bizId);
 		}
 	}
+
+	/**
+	 * Returns the current session/conversation user as an Admin module User
+	 *
+	 * @return The current {@link modules.admin.User.UserExtension}
+	 */
+	@SuppressWarnings("static-method")
+	public UserExtension currentAdminUser() {
+		UserExtension result = null;
+		try {
+			Persistence p = CORE.getPersistence();
+			result = p.retrieve(User.MODULE_NAME,
+					User.DOCUMENT_NAME,
+					p.getUser().getId());
+		} catch (@SuppressWarnings("unused") Exception e) {
+			// do nothing
+		}
+
+		return result;
+	}
+
+	/**
+	 * Returns the current session/conversation user as an Admin module UserProxy
+	 *
+	 * @return The current {@link modules.admin.domain.UserProxy}
+	 */
+	@SuppressWarnings("static-method")
+	public UserProxyExtension currentAdminUserProxy() {
+		UserProxyExtension result = null;
+		try {
+			Persistence p = CORE.getPersistence();
+			result = p.retrieve(UserProxy.MODULE_NAME, UserProxy.DOCUMENT_NAME, p.getUser().getId());
+		} catch (@SuppressWarnings("unused") Exception e) {
+			// do nothing
+		}
+
+		return result;
+	}
+
+	/**
+	 * Creates a new admin User for a given contact
+	 * - sets the new user name to be the contact email address
+	 * - adds the specified group privileges to the user
+	 * - sets their home Module (if provided)
+	 * - sets an expired password (to force them to reset their password)
+	 * - sets a password reset token that can be provided to the user to reset their password
+	 * - optionally sends an invitation email
+	 *
+	 * @param contact The Contact to create the new User from
+	 * @param groupName The name of the group to assign to the user
+	 * @param homeModuleName The default module to display after login
+	 * @param sendInvitation Whether to send an invitation email to the new user
+	 * @return The newly created UserExtension instance
+	 */
+	@SuppressWarnings("static-method")
+	public UserExtension createAdminUserFromContactWithGroup(ContactExtension contact, final String groupName,
+			final String homeModuleName, final boolean sendInvitation) {
+
+		if (contact == null) {
+			throw new DomainException("admin.modulesUtils.createAdminUserFromContactWithGroup.exception.contact");
+		}
+
+		if (groupName == null) {
+			throw new DomainException("admin.modulesUtils.createAdminUserFromContactWithGroup.exception.groupName");
+		}
+
+		// check if user already exists
+		DocumentQuery q = CORE.getPersistence()
+				.newDocumentQuery(User.MODULE_NAME, User.DOCUMENT_NAME);
+		q.getFilter().addEquals(User.userNamePropertyName, contact.getEmail1());
+		q.setMaxResults(1);
+
+		UserExtension found = q.beanResult();
+		if (found != null) {
+			throw new DomainException("admin.modulesUtils.createAdminUserFromContactWithGroup.exception.duplicateUser");
+		}
+
+		// check the group exists
+		DocumentQuery qGroup = CORE.getPersistence().newDocumentQuery(Group.MODULE_NAME, Group.DOCUMENT_NAME);
```

Comment:
Persistence should be injected in service classes. 

Should there be a `GroupService` to retrieve the group? This would be a better pattern for other developers to follow.